rotkehlchen/exchanges/binance.py

```diff
@@ -680,6 +681,73 @@ def _query_margined_futures_balances(
 
         return balances
 
+    def _query_pools_balances(
+            self,
+            balances: DefaultDict[Asset, Balance],
+    ) -> DefaultDict[Asset, Balance]:
+
+        def process_pool_asset(asset_name: str, asset_amount: FVal) -> None:
+
+            if asset_amount == ZERO:
+                return None
+
+            try:
+                asset = asset_from_binance(asset_name)
+            except UnsupportedAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found unsupported {self.name} asset {asset_name}. '
+                    f'Ignoring its margined futures balance query. {str(e)}',
+                )
+                return None
+            except UnknownAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found unknown {self.name} asset {asset_name}. '
+                    f'Ignoring its margined futures balance query. {str(e)}',
+                )
+                return None
+            except DeserializationError as e:
+                log.error(
+                    f'{self.name} balance deserialization error ',
+                    f'with asset {asset_name}. {str(e)}',
+                )
+                return None
+
+            try:
+                usd_price = Inquirer().find_usd_price(asset)
+            except RemoteError as e:
+                self.msg_aggregator.add_error(
+                    f'Error processing {self.name} balance entry due to inability to '
+                    f'query USD price: {str(e)}. Skipping margined futures balance entry',
+                )
+                return None
+
+            balances[asset] += Balance(
+                amount=asset_amount,
+                usd_value=asset_amount * usd_price,
+            )
+
+            return None
+
+        try:
+            response = self.api_query('sapi', 'bswap/liquidity')
+        except RemoteError as e:
+            log.warning(
+                f'Insufficient permission to query {self.name} pool balances.'
+                f'Skipping query. Response details: {str(e)}',
+            )
+            return balances
+
+        try:
+            for entry in response:
+                for asset_name, asset_amount in entry['share']['asset'].items():
+                    process_pool_asset(asset_name, FVal(asset_amount))
+        except KeyError as e:
+            log.error(
+                f'At {self.name} pool balances did not find expected key {str(e)}',
+            )
```

Comment:
1. You need to notify the user that an error happened here, so separate logging from msg_aggregator.add_error
2. If entry['share']['asset'] is a string or an integer this will raise an `AttributeError` that will go unhandled and kill the thread.

```suggestion
        try:
            for entry in response:
                for asset_name, asset_amount in entry['share']['asset'].items():
                    process_pool_asset(asset_name, FVal(asset_amount))
        except (KeyError, AttributeError) as e:
            self.msg_aggregator.add_error(
                f'At {self.name} pool balances got unexpected data format. Skipping them in the balance query. Check logs for details',
            )
            if isinstance(e, KeyError):
                msg = 'Missing key {str(e)}'
            else:
                msg = str(e)
            log.error(f'Unexpected data format returned by {self.name} pools. Data: {response). Error: {msg}')
```